lib/srv/db/spanner/grpcserver.go

```diff
@@ -0,0 +1,467 @@
+}
+
+func (e *Engine) Commit(ctx context.Context, req *spannerpb.CommitRequest) (*spannerpb.CommitResponse, error) {
+	if err := e.authorizeRPCRequest(ctx, req.GetSession(), "Commit", req, false); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clt, err := e.getClient(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	res, err := clt.Commit(ctx, req)
+	return res, trace.Wrap(err)
+}
+
+func (e *Engine) Rollback(ctx context.Context, req *spannerpb.RollbackRequest) (*emptypb.Empty, error) {
+	if err := e.authorizeRPCRequest(ctx, req.GetSession(), "Rollback", req, false); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clt, err := e.getClient(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	res, err := clt.Rollback(ctx, req)
+	return res, trace.Wrap(err)
+}
+
+func (e *Engine) PartitionRead(ctx context.Context, req *spannerpb.PartitionReadRequest) (*spannerpb.PartitionResponse, error) {
+	if err := e.authorizeRPCRequest(ctx, req.GetSession(), "PartitionRead", req, false); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clt, err := e.getClient(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	res, err := clt.PartitionRead(ctx, req)
+	return res, trace.Wrap(err)
+}
+
+func (e *Engine) BatchWrite(req *spannerpb.BatchWriteRequest, stream spannerpb.Spanner_BatchWriteServer) error {
+	ctx := stream.Context()
+	if err := e.authorizeRPCRequest(ctx, req.GetSession(), "BatchWrite", req, false); err != nil {
+		return trace.Wrap(err)
+	}
+	clt, err := e.getClient(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	cc, err := clt.BatchWrite(ctx, req)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return trace.Wrap(proxyServerStream(cc, stream))
+}
+
+func (e *Engine) ExecuteSql(ctx context.Context, req *spannerpb.ExecuteSqlRequest) (*spannerpb.ResultSet, error) {
+	skipAudit := isQuerySpam(req.GetSql())
+	if err := e.authorizeRPCRequest(ctx, req.GetSession(), "ExecuteSql", req, skipAudit); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clt, err := e.getClient(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	res, err := clt.ExecuteSql(ctx, req)
+	return res, trace.Wrap(err)
+}
+
+func (e *Engine) PartitionQuery(ctx context.Context, req *spannerpb.PartitionQueryRequest) (*spannerpb.PartitionResponse, error) {
+	if err := e.authorizeRPCRequest(ctx, req.GetSession(), "PartitionQuery", req, false); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	clt, err := e.getClient(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	res, err := clt.PartitionQuery(ctx, req)
+	return res, trace.Wrap(err)
+}
+
+// isQuerySpam returns true if a query is just some spam not worth audit logging.
+func isQuerySpam(query string) bool {
+	// obviously this is a very basic heuristic, but GUI clients in particular
+	// love to spam these.
+	query = strings.ToLower(strings.TrimSpace(query))
+	switch query {
+	case "select 1":
+	case "select 'keep alive'":
+	default:
+		return false
+	}
+	return true
+}
```

Comment:
a long time ago I remember seeing some code where we filter events. I can't remember where it is, somewhere deep in our emitter or recorder code I think.
I'll try to find that for some prior art, maybe we can do something similar. 

I also like the idea of just filtering the view in the web ui, but that just means anyone exporting events to their own SIEM will also get spammed with events. I guess that at least puts the power in their hands, but I think seeing "SELECT 1" is low value... we could also just try to deduplicate them.

I do have the same apprehension that you have though - actually removed this spam filter once for the same concern, and then added it back because the spam was so annoying.

Reply:
my take on this is it would be difficult to categorize what's a "spam". GUI tools can change their "spam" queries at any time. "select 1" is our favourite query for testing connections during manual testing so I would be surprised if I could not find it in my audit log. Imagine during a debug session with a customer, we have to tell them to use "select 2" but not "select 1".

Reply:
removed the filter in 033315cb2432bdbac31d3c55d1fb775fa91b27e1